### PR TITLE
reposilitePlugins: init at 3.5.19

### PR DESCRIPTION
**Pull request by uku3lig**
This is the first of a few PR's in preparation for the addition of a nixos/reposilite module.

This pr simply adds the 5 official reposilite plugins, packaged as jar files.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/re/reposilite/plugins.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    mkPlugins = lib.mapAttrs (
+      name: hash:
+      self.callPackage (
+        { fetchurl }:
+        fetchurl {
+          url = "https://maven.reposilite.com/releases/com/reposilite/plugin/${name}-plugin/${version}/${name}-plugin-${version}-all.jar";
+          inherit version hash;
+
+          meta = {
+            description = "${name} plugin for Reposilite.";
```
Comment (pluiedev) [inline pkgs/by-name/re/reposilite/plugins.nix:20]: This would make it so that the description would start with a lowercase character which is not ideal

**Review by pluiedev — CHANGES_REQUESTED** [review]

File: pkgs/by-name/re/reposilite/plugins.nix (hunk 1 above)
Comment (uku3lig) [reply pkgs/by-name/re/reposilite/plugins.nix]: hopefully resolved by the ugly thing i wrote

File: pkgs/by-name/re/reposilite/plugins.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    );
+  in
+  mkPlugins {
+    checksum = "sha256-SOW+J5D0YYz3uTYZUV5ScSY1Sg3Acm0hymLI9R+xQ9Y=";
+    groovy = "sha256-/YPhyMzzKT3ncH1tYZA3hBV+KcnmB590e1//rt2mPu4=";
+    migration = "sha256-DSkrtrySoi5Y0EFhIDqkZ64eQQGgdWRKNHy9IKLTQr8=";
+    prometheus = "sha256-gEomMIxJBC9y1Z3V4X/nrbXAwufYH+4uCZAdOBjDuo4=";
+    swagger = "sha256-qOADxWfCYw8xOE26+povSyRpiPs6u3q9Ro2XK7VAlJY=";
+  }
+)
```
Comment (getchoo) [inline pkgs/by-name/re/reposilite/plugins.nix:44]:
```suggestion
      )
    ) { };
  }
  // lib.mapAttrs (name: hash: self.fetchPlugin { inherit name hash; }) {
    checksum = "sha256-SOW+J5D0YYz3uTYZUV5ScSY1Sg3Acm0hymLI9R+xQ9Y=";
    groovy = "sha256-/YPhyMzzKT3ncH1tYZA3hBV+KcnmB590e1//rt2mPu4=";
    migration = "sha256-DSkrtrySoi5Y0EFhIDqkZ64eQQGgdWRKNHy9IKLTQr8=";
    prometheus = "sha256-gEomMIxJBC9y1Z3V4X/nrbXAwufYH+4uCZAdOBjDuo4=";
    swagger = "sha256-qOADxWfCYw8xOE26+povSyRpiPs6u3q9Ro2XK7VAlJY=";
  };
}
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Review by getchoo — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355505`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>reposilitePlugins.checksum</li>
    <li>reposilitePlugins.groovy</li>
    <li>reposilitePlugins.migration</li>
    <li>reposilitePlugins.prometheus</li>
    <li>reposilitePlugins.swagger</li>
  </ul>
</details>

**Comment by getchoo** [comment]
@ofborg eval

**Review by getchoo — APPROVED** [review]

File: pkgs/by-name/re/reposilite/plugins.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
+  lib,
+  newScope,
+  reposilite,
+}:
+lib.makeScope newScope (
```
[the remark this answers is not in the document]
Comment (Artturin) [reply pkgs/by-name/re/reposilite/plugins.nix]: @uku3lig You overlooked the `"reposilitePlugins"` so this broke cross eval.

[the remark this answers is not in the document]
Comment (uku3lig) [reply pkgs/by-name/re/reposilite/plugins.nix]: @artturin oh, my apologies! stuff seemed to be working fine when i tested with nix-build -A, i hope this isn't too big of an issue

**Pull request merged**